src/main/java/org/peergos/HttpProxyService.java

```diff
@@ -24,22 +25,30 @@ public class HttpProxyService {
     private static final Logger LOG = Logging.LOG();
     private final Host node;
     private final HttpProtocol.Binding p2pHttpBinding;
+    private final Kademlia dht;
     public static final String API_URL = "/p2p/";
 
-    public HttpProxyService(Host node, HttpProtocol.Binding p2pHttpBinding) {
+    public HttpProxyService(Host node, HttpProtocol.Binding p2pHttpBinding, Kademlia dht) {
         this.node = node;
         this.p2pHttpBinding = p2pHttpBinding;
+        this.dht = dht;
     }
     public ProxyResponse proxyRequest(Multihash targetNodeId, ProxyRequest request) throws IOException {
 
         AddressBook addressBook = node.getAddressBook();
-        Optional<Multiaddr> targetAddressesOpt = addressBook.get(PeerId.fromBase58(targetNodeId.toBase58())).join().stream().findFirst();
+        Optional<Multiaddr> targetAddressesOpt = addressBook.get(PeerId.fromBase58(targetNodeId.bareMultihash().toBase58())).join().stream().findFirst();
         if (targetAddressesOpt.isEmpty()) {
-            LOG.info("Target not found in address book: " + targetNodeId);
-            return new ProxyResponse(new byte[0], new HashMap<>(), 404);
+            List<PeerAddresses> closestPeers = dht.findClosestPeers(targetNodeId, 20, node);
+            Optional<PeerAddresses> matching = closestPeers.stream().filter(p -> p.peerId.equals(targetNodeId)).findFirst();
+            if (matching.isEmpty()) {
+                LOG.info("Target not found: " + targetNodeId);
+                return new ProxyResponse(new byte[0], new HashMap<>(), 404);
```

Comment:
I think this should error. Otherwise we can't tell the difference between reaching the server and it returning 404 and not finding the server

Reply:
done